---
title: Learning About Python
author: Michaela Priess
date: '2019-12-15'
slug: python-fun
categories: []
tags: []
description: ''
---



<div id="python-in-genetics" class="section level2">
<h2>Python in Genetics</h2>
<p>Prior to my participation in this class, I had never before worked with Python. While it was (and honestly still is) intimidating at first, I have slowly been getting more comfortable working with Python. I can now definitely see how proficiency in Python is a useful tool to have in computational biology/bioinformatics. Since my degree in biology specifically focuses on genetics and genomics, it was particularly interesting to me when we used regular expressions in python to find endonuclease target sites within a sequence of DNA. As can be seen below, in this case I used re.findall() to identify 4 different endonuclease target sites within the given DNA sequence. Besides just identifying the total number of sites within the sequence, I was able to see what the actual sequence at each identified site was.</p>
<pre class="r"><code>library(reticulate)</code></pre>
<pre class="python"><code>import re

string7=&quot;ATGGCAATAACCCCCCGTTTCTACTTCTAGAGGAGAAAAGTATTGACATGAGCGCTCCCGGCACAAGGGCCAAAGAAGTCTCCAATTTCTTATTTCCGAATGACATGCGTCTCCTTGCGGGTAAATCACCGACCGCAATTCATAGAAGCCTGGGGGAACAGATAGGTCTAATTAGCTTAAGAGAGTAAATCCTGGGATCATTCAGTAGTAACCATAAACTTACGCTGGGGCTTCTTCGGCGGATTTTTACAGTTACCAACCAGGAGATTTGAAGTAAATCAGTTGAGGATTTAGCCGCGCTATCCGGTAATCTCCAAATTAAAACATACCGTTCCATGAAGGCTAGAATTACTTACCGGCCTTTTCCATGCCTGCGCTATACCCCCCCACTCTCCCGCTTATCCGTCCGAGCGGAGGCAGTGCGATCCTCCGTTAAGATATTCTTACGTGTGACGTAGCTATGTATTTTGCAGAGCTGGCGAACGCGTTGAACACTTCACAGATGGTAGGGATTCGGGTAAAGGGCGTATAATTGGGGACTAACATAGGCGTAGACTACGATGGCGCCAACTCAATCGCAGCTCGAGCGCCCTGAATAACGTACTCATCTCAACTCATTCTCGGCAATCTACCGAGCGACTCGATTATCAACGGCTGTCTAGCAGTTCTAATCTTTTGCCAGCATCGTAATAGCCTCCAAGAGATTGATGATAGCTATCGGCACAGAACTGAGACGGCGCCGATGGATAGCGGACTTTCGGTCAACCACAATTCCCCACGGGACAGGTCCTGCGGTGCGCATCACTCTGAATGTACAAGCAACCCAAGTGGGCCGAGCCTGGACTCAGCTGGTTCCTGCGTGAGCTCGAGACTCGGGATGACAGCTCTTTAAACATAGAGCGGGGGCGTCGAACGGTCGAGAAAGTCATAGTACCTCGGGTACCAACTTACTCAGGTTATTGCTTGAAGCTGTACTATTTTAGGGGGGGAGCGCTGAAGGTCTCTTCTTCTCATGACTGAACTCGCGAGGGTCGTGAAGTCGGTTCCTTCAATGGTTAAAAAACAAAGGCTTACTGTGCGCAGAGGAACGCCCATCTAGCGGCTGGCGTCTTGAATGCTCGGTCCCCTTTGTCATTCCGGATTAATCCATTTCCCTCATTCACGAGCTTGCGAAGTCTACATTGGTATATGAATGCGACCTAGAAGAGGGCGCTTAAAATTGGCAGTGGTTGATGCTCTAAACTCCATTTGGTTTACTCGTGCATCACCGCGATAGGCTGACAAAGGTTTAACATTGAATAGCAAGGCACTTCCGGTCTCAATGAACGGCCGGGAAAGGTACGCGCGCGGTATGGGAGGATCAAGGGGCCAATAGAGAGGCTCCTCTCTCACTCGCTAGGAGGCAAATGTAAAACAATGGTTACTGCATCGATACATAAAACATGTCCATCGGTTGCCCAAAGTGTTAAGTGTCTATCACCCCTAGGGCCGTTTCCCGCATATAAACGCCAGGTTGTATCCGCATTTGATGCTACCGTGGATGAGTCTGCGTCGAGCGCGCCGCACGAATGTTGCAATGTATTGCATGAGTAGGGTTGACTAAGAGCCGTTAGATGCGTCGCTGTACTAATAGTTGTCGACAGACCGTCGAGATTAGAAAATGGTACCAGCATTTTCGGAGGTTCTCTAACTAGTATGGATTGCGGTGTCTTCACTGTGCTGCGGCTACCCATCGCCTGAAATCCAGCTGGTGTCAAGCCATCCCCTCTCCGGGACGCCGCATGTAGTGAAACATATACGTTGCACGGGTTCACCGCGGTCCGTTCTGAGTCGACCAAGGACACAATCGAGCTCCGATCCGTACCCTCGACAAACTTGTACCCGACCCCCGGAGCTTGCCAGCTCCTCGGGTATCATGGAGCCTGTGGTTCATCGCGTCCGATATCAAACTTCGTCATGATAAAGTCCCCCCCTCGGGAGTACCAGAGAAGATGACTACTGAGTTGTGCGAT&quot;

print re.findall(r&quot;(A.TAAT)|(GC(A|G)(A|T)TG)&quot;, string7)</code></pre>
<pre><code>## [(&#39;&#39;, &#39;GCGTTG&#39;, &#39;G&#39;, &#39;T&#39;), (&#39;ATTAAT&#39;, &#39;&#39;, &#39;&#39;, &#39;&#39;), (&#39;&#39;, &#39;GCAATG&#39;, &#39;A&#39;, &#39;A&#39;), (&#39;ACTAAT&#39;, &#39;&#39;, &#39;&#39;, &#39;&#39;)]</code></pre>
<p>In addition to using regular expressions in python to identify specificed sequences, I also learned how functions in python can be used to quickly analyze DNA sequences. For instance, a function can be used to identify all k-mers, or sequences of specified length k, within a given DNA sequence. An example of this function and its output can be seen below.</p>
<div class="figure">
<img src="/img/python_example.png" alt="Example of Python k-mer Function" />
<p class="caption">Example of Python k-mer Function</p>
</div>
</div>
